lib/models/candle/audit_gaps.js

```diff
@@ -0,0 +1,26 @@
+const { TIME_FRAME_WIDTHS } = require('bfx-hf-util')
+const _isFinite = require('lodash/isFinite')
+
+module.exports = ({ getInRange }) => (doc, { start, end }) => {
+  const gaps = []
+  const { tf } = doc
+  const candles = getInRange(doc, { start, end })
+
+  if (candles.length < 2) {
+    return { gaps, candles }
+  }
+
+  const width = TIME_FRAME_WIDTHS[tf]
+
+  if (!_isFinite(width)) {
+    throw new Error(`invalid time frame [unknown width]: ${tf}`)
+  }
+
+  for (let i = 0; i < candles.length - 1; i += 1) {
+    if ((candles[i].mts - candles[i + 1].mts) !== width) {
+      gaps.push(i)
```

Comment:
We could improve the efficiency of this later on by stitching together gaps if there are less than X candles inbetween. For example we can get 4k candles in a single request so if we can cover multiple gaps in a single 4k request then we could just forget about the candles we have inbetween.

Reply:
Good point, but keep in mind this is meant for internal generic use to ensure there is a candle for each point in time in a range. The req limit is external to this logic; we can always add a `width` arg in the future to override the internal width.